private/pkg/observability/observabilityotel/tracer_provider_closer.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observabilityotel
+
+import (
+	"context"
+
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type tracerProviderCloser struct {
+	tracerProvider *sdktrace.TracerProvider
+}
+
+func newTracerProviderCloser(tracerProvider *sdktrace.TracerProvider) *tracerProviderCloser {
+	return &tracerProviderCloser{
+		tracerProvider: tracerProvider,
+	}
+}
+
+func (t *tracerProviderCloser) Close() error {
+	// TODO: this should probably be passed down from the application entrypoint
```

Comment:
Don't leave hanging TODOs -- associate it with an issue to actually fix it, eg `// TODO (INF-123): fix this coz of a, b, and c.`, or `// TODO (#1234): blabla`. The issue can carry more dynamic context.

FWIW, I'm not sure if we need this TODO. I'd instead leave a `ctx Context` in the arguments, and wrap the closing for `chainCloser` - if this package is "self contained", it should also forward that context, but the way in which we _use_ it is where we lose the context (eg we don't have a `stop observability` endpoint that'd have a request-associated contexT)

Reply:
After an offline discussion looking at the various ways we can do this, we've decided to annotate this clearly with a comment and will support a `context.Context` in the layer above if this is every required.